### PR TITLE
1278 Venn Diagram of chemicals per group type

### PR DESCRIPTION
**Pull request by mwfrost**
Closes: #1278 

There is some craziness going on in this branch. 

First, I have created a new app called `etl` for loading and maintaining ad hoc queries that we don't want to calculate at runtime. Migrate that app up to 0002 to load the model object, then run this to create its table:
```
from django.apps import apps                                                                                                   
SummarySQL = apps.get_model("etl", "SummarySQL")                                                                               
e = SummarySQL.objects.filter(table_name="summary_sid_per_grouptype").first()  
e.refresh()  
```
The table is a very stripped-down combination of bit strings and chemical counts - it gets turned into json for the Venn diagram in an ajax endpoint:
http://127.0.0.1:8000/sid_gt_json

### CODE REVIEW

**Comment by mcovalt** [comment]
This is very cool. Do we want to integrate this into the Celery scheduler? If so, we could do the following:

In "etl/tasks.py"
```python
from celery import shared_task
from etl.models import SummarySQL

@shared_task
def refresh_summary():
    SummarySQL.refresh_all()
```

In "factotum/celery.py":
```python
...
from etl.tasks import refresh_summary
...

@app.on_after_configure.connect
def setup_periodic_tasks(sender, **kwargs):
    # Calls refresh_summary() every hour.
    sender.add_periodic_task(360.0, refresh_summary.s(), name='add every 10')
```

**Comment by mwfrost** [comment]
@mcovalt Yes, that was my plan. I have stubbed in just enough that we can call it from celery later.

**Comment by michael-barbour** [comment]
In order to review this, I think an overview on where you landed on scope of the venn diagram is required.
From the ticket: 

> Also possible that we could/may want to limit the Venn diagram to only a smaller number of specific group types (probably the three 'main' group types, e.g. LP, FU, CO)

This seems to not be reflected in the code.  It appears that even with valid data this plot isn't entirely correct.  The below image represents this [json](https://github.com/HumanExposure/factotum/files/4047993/sid_gt_json.txt).  Of note, my Chemical Presense is entirely contained within the [Composition/Chemical Presense] and [Composition/HHE/Chemical Presense] sets yet [Chemical Presense] and [Chemical Presense/HHE] are both visible (although the mouse-over text does not reflect that).

image

**Comment by mwfrost** [comment]
I have changed the ETL app so that it no longer creates any tables in the database. Now the results of each SummarySQL query are stored in the `results` field of that model as JSON-formatted text. The `ajax.py` view gets its data from the SummarySQL model rather than looking in the database. The changes should be totally transparent to Celery.

**Comment by mwfrost** [comment]
@michael-barbour That example has me wondering if it's even possible to draw an accurate diagram with this library. The diagram would look like this: 
IMG_0580

**Comment by mwfrost** [comment]
@michael-barbour I plugged those example numbers into the demo tool on the developer's website: http://www.benfrederickson.com/venn-diagrams-with-d3.js/

It fails in the same way.
Screen Shot 2020-01-14 at 7 04 23 AM

**Comment by mwfrost** [comment]
Matt Covalt has removed the ETL app entirely and now the expensive query is being performed through the ORM in a `@cache_page` view
